### DISCUSSION TITLE
XplatGenerateReleaseNotes tunneling socket could not be established

### DISCUSSION THREAD

**Discussion by vityok87** [rfennell/AzurePipelines#1047] [Q&A]
Hello,

Trying to figure out why the XplatGenerateReleaseNotes extension cannot connect to the Azure DevOps Server.

My Setup:

- Self hosted Azure Pipelines Agent v2.188.4
- Generate Release Notes (Crossplatform) version 3.57.2
- Self hosted Agent is run on a docker container
- HTTP_PROXY, HTTPS_PROXY and NO_PROXY are passed in as ENV variables to the container
- Tried setting pipeline variables (assumed this to be the issue) but does not change the error
- Tried using the PAT instead of the OAUTH token
```
 > variables:
 >   System.Debug: true
 >   NODE_TLS_REJECT_UNAUTHORIZED: 0
```
- Other plugins/extensions do not have this issue.
- The agent has no issues connecting to ADO server to download repos
- Tested this on my machine and also on the company servers

> ```
> Creating Azure DevOps API connections for https://**************/ with 'allowRetries' set to 'true' and 'maxRetries' count to '20'
> 2021-07-21T16:46:30.3220538Z ##[debug]Getting System.AccessToken
> 2021-07-21T16:46:30.3221073Z ##[debug]Getting credentials the agent is running as
> 2021-07-21T16:46:30.3224200Z ##[debug]SYSTEMVSSCONNECTION exists true
> `##[debug]Found an OAUTH token`
> `Creating the credential handler from the OAUTH token`
> `(node:1139) UnhandledPromiseRejectionWarning: Error: tunneling socket could not be established, statusCode=502`
> `at ClientRequest.onConnect (/azp/agent/_work/_tasks/XplatGenerateReleaseNotes_7b0a5887-75c4-4ade-a915-f92a93c934cb/3.57.2/node_modules/tunnel/lib/tunnel.js:154:19)`
> at Object.onceWrapper (events.js:286:20)
> at ClientRequest.emit (events.js:198:13)
> at Socket.socketOnData (_http_client.js:484:11)
> at Socket.emit (events.js:198:13)
> at addChunk (_stream_readable.js:288:12)
> at readableAddChunk (_stream_readable.js:269:11)
> at Socket.Readable.push (_stream_readable.js:224:10)
> at TCP.onStreamRead (internal/stream_base_commons.js:94:17)
> (node:1139) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
> (node:1139) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
> ```

Any help would be greatly appreciated

**Comment by rfennell** [comment]
The main difference between this extension and many other are the Azure DevOps API calls it makes. So that is probably the best area to focus on, but we need to try to isolate the problem a bit more. 

So a couple of things to try to help diagnose (but not fix)
- If you use the `overridePat` parameter to inject PAT to access the API as opposed to the Pipeline Token does this make any difference (in case the issue is with accessing the Pipeline Token)
- If you run a private agent on a PC/VM i.e. not in the container do you get the same error (again with the Agent Token and the   `overridePat` parameter to inject PAT)
